[RELEASE-v1.4] Support config to deploy internal certificates automatically (#13005)

nak3 opened the pull request:
This patch cherry-pics https://github.com/knative/serving/commit/58cce5440428e7e1671106ba0951538e378dc890.

skonto commented:
@nak3 in order for tests to run properly since we merged first https://github.com/openshift/knative-serving/pull/1173 we need to also merge https://github.com/openshift-knative/serverless-operator/pull/1642. Ideally the latter needs to get in first.

skonto commented:
/retest

nak3 commented:
/retest

skonto commented:
```
oc create configmap kourier-cm -n openshift-serverless --from-file=/tmp/kourier.yaml
error: failed to create configmap: namespaces "openshift-serverless" not found
```
/retest

nak3 commented:
/test 49-e2e-aws-ocp-49

nak3 commented:
@skonto This PR is ready to go now. Could you please take a look this and https://github.com/openshift-knative/net-kourier/pull/23?

(https://github.com/openshift/knative-serving/pull/1188 also verified with enabling internal-encryption .)

skonto commented:
Same error?
```
+ oc create configmap kourier-cm -n openshift-serverless --from-file=/tmp/kourier.yaml
error: failed to create configmap: namespaces "openshift-serverless" not found
```

nak3 commented:
Yeah, I will look into it tomorrow.
Hmm.. why only 4.9...

skonto commented:
Creation of the ns failed earlier seems infra.

nak3 commented:
Ah, you are right.

```
+ oc create namespace openshift-serverless
The connection to the server api.ci-ocp-4-9-amd64-aws-us-west-1-pv5gz.hive.aws.ci.openshift.org:6443 was refused - did you specify the right host or port?
```

skonto commented:
@nak3 there is a file added: `[cmd/activator/.main.go.swp](https://github.com/openshift/knative-serving/pull/1183/files#diff-3fcaeec44c6f0b76f24357ba5b1d6a76f9c6fb3b3243dfdc69b50665eaec16fd)`

nak3 commented:
No, I removed it. It was added by mistake on upstream and removed https://github.com/knative/serving/commit/92b11a8826f151728ac0a0a936def5123bedd99e .
I think we should remove it here too.

skonto commented:
Yes we should remove it its garbage.

skonto commented:
/test 49-e2e-aws-ocp-49

skonto commented:
@nak3 this seems ready. LGTM but can we remove the swp file?

skonto commented:
@nak3 let's remove the file and then we will be ready, all tests pass.

nak3 commented:
Nah, this PR is removing the swap file. It will be removed when you merged this PR.

skonto commented:
Ok my bad misread. What about the order of merging should we do the S-O one first?

nak3 commented:
Either is fine. So. I updated S-O for ready.

skonto commented:
/lgtm
